teuthology/test/test_run.py

```diff
@@ -83,7 +83,7 @@ def test_validate_task_no_tasks(self):
         assert result == []
 
     def test_validate_tasks_valid(self):
-        expected = {"foo": "bar"}
```

Comment:
We should have another test where you pass in a dict and assert that an AssertionError is raised.  We have a test below for something similar (test_get_invalid_tasks_invalid), but now instead of making sure targets isn't in the list it's failing on it being a dict and not a list.  Two different failure conditions there.